Fails to build for me on linux. I don't think it's an oom. Maybe it needs openimageio v2 after all? Or some other change

```
/nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: libusd_ms.so: undefined reference to `OSL_v1_13::OSLQuery::open_bytecode(OpenImageIO_v3_0::basic_string_view<char, std::char_traits<char> >)'
/nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: libusd_ms.so: undefined reference to `OpenImageIO_v3_0::ustring::empty_std_string[abi:cxx11]'
/nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: libusd_ms.so: undefined reference to `OpenImageIO_v3_0::TypeDesc::c_str() const'
/nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: libusd_ms.so: undefined reference to `OSL_v1_13::OSLQuery::open(OpenImageIO_v3_0::basic_string_view<char, std::char_traits<char> >, OpenImageIO_v3_0::basic_string_view<char, std::char_traits<char> >)'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

[openusd-x86_64-linux.log](https://github.com/user-attachments/files/20273250/openusd-x86_64-linux.log)

Also, I haven't worked with c++ in a while, so could you give some context for the `opensubdiv` change? Maybe it's completely obvious to others, and this is not a blocker, but it's not fully clear to me what it does and why we need it

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 408041`

---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>openusd</li>
    <li>python312Packages.openusd</li>
    <li>python313Packages.openusd</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>blender</li>
    <li>blender-hip</li>
    <li>blendfarm</li>
    <li>opensubdiv</li>
    <li>opensubdiv.dev</li>
    <li>opensubdiv.static</li>
  </ul>
</details>

--- a/pkgs/by-name/op/opensubdiv/cmake-config.patch
+++ b/pkgs/by-name/op/opensubdiv/cmake-config.patch
@@ -1,0 +1,13 @@
+diff --git a/opensubdiv-config.cmake.in b/opensubdiv-config.cmake.in
+index 08a005aa..48fd2cdc 100644
+--- a/opensubdiv-config.cmake.in
++++ b/opensubdiv-config.cmake.in
+@@ -1,6 +1,6 @@
+ @PACKAGE_INIT@
+ 
+-set_and_check(OpenSubdiv_INCLUDE_DIR "${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_INCLUDEDIR@")
+-set_and_check(OpenSubdiv_LIB_DIR "${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_LIBDIR@")
++set(OpenSubdiv_INCLUDE_DIR "@CMAKE_INSTALL_FULL_INCLUDEDIR@")
++set(OpenSubdiv_LIB_DIR "@CMAKE_INSTALL_FULL_LIBDIR@")
+ 
+ include("${CMAKE_CURRENT_LIST_DIR}/OpenSubdivTargets.cmake")

--- a/pkgs/by-name/op/opensubdiv/package.nix
+++ b/pkgs/by-name/op/opensubdiv/package.nix
@@ -42,6 +42,7 @@ stdenv.mkDerivation (finalAttrs: {
     ++ lib.optionals cudaSupport [
       cudaPackages.cuda_nvcc
     ];
+
   buildInputs =
     lib.optionals stdenv.hostPlatform.isUnix [
       libGLU
@@ -55,11 +56,17 @@ stdenv.mkDerivation (finalAttrs: {
       xorg.libXinerama
       xorg.libXi
     ]
-    ++ lib.optionals (openclSupport && stdenv.hostPlatform.isLinux) [ ocl-icd ]
-
+    ++ lib.optionals (openclSupport && stdenv.hostPlatform.isLinux) [
+      ocl-icd
+    ]
     ++ lib.optionals cudaSupport [
       cudaPackages.cuda_cudart
     ];
+
+  patches = [
+    # Prevent CMake from generating a redundant nested path like /nix/store/.../nix/store/...
+    ./cmake-config.patch
+  ];
 
   # It's important to set OSD_CUDA_NVCC_FLAGS,
   # because otherwise OSD might piggyback unwanted architectures:
@@ -105,8 +112,14 @@ stdenv.mkDerivation (finalAttrs: {
       ''
     else
       ''
-        moveToOutput "lib/*.a" $static
+        moveToOutput "lib/libosd*.a" $static
       '';
+
+  postFixup = ''
+    # Adjust static library path to reflect relocation to $static
+    sed -i -E "s|\\\$\{_IMPORT_PREFIX\}/lib/(libosd.*\.a)|$static/lib/\1|" \
+      $dev/lib/cmake/OpenSubdiv/OpenSubdivTargets-release.cmake
+  '';
 
   meta = {
     description = "Open-Source subdivision surface library";

--- a/pkgs/development/compilers/osl/default.nix
+++ b/pkgs/development/compilers/osl/default.nix
@@ -15,6 +15,7 @@
   bison,
   partio,
   pugixml,
+  robin-map,
   util-linux,
   python3,
 }:
@@ -24,13 +25,13 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "openshadinglanguage";
-  version = "1.13.11.0";
+  version = "1.14.5.1";
 
   src = fetchFromGitHub {
     owner = "AcademySoftwareFoundation";
     repo = "OpenShadingLanguage";
     rev = "v${version}";
-    hash = "sha256-E/LUTtT0ZU0SBuwtJPA0FznvOuc2a3aJn2/n3ru5l0s=";
+    hash = "sha256-dmGVCx4m2bkeKhAJbU1mrzEDAmnL++7GA5okb9wwk/Y=";
   };
 
   cmakeFlags = [
@@ -43,10 +44,12 @@ stdenv.mkDerivation rec {
     "-DLLVM_DIRECTORY=${llvm}"
     "-DLLVM_CONFIG=${llvm.dev}/bin/llvm-config"
     "-DLLVM_BC_GENERATOR=${clang}/bin/clang++"
-
-    # Set C++11 to C++14 required for LLVM10+
-    "-DCMAKE_CXX_STANDARD=14"
   ];
+
+  prePatch = ''
+    substituteInPlace src/cmake/modules/FindLLVM.cmake \
+      --replace-fail "NO_DEFAULT_PATH" ""
+  '';
 
   preConfigure = "patchShebangs src/liboslexec/serialize-bc.bash ";
 
@@ -67,6 +70,7 @@ stdenv.mkDerivation rec {
       partio
       pugixml
       python3.pkgs.pybind11
+      robin-map
       util-linux # needed just for hexdump
       zlib
     ]

--- a/pkgs/development/python-modules/openusd/default.nix
+++ b/pkgs/development/python-modules/openusd/default.nix
@@ -23,7 +23,7 @@
   ninja,
   numpy,
   opencolorio,
-  openimageio_2,
+  openimageio,
   opensubdiv,
   osl,
   ptex,
@@ -118,7 +118,7 @@ buildPythonPackage rec {
       imath
       materialx
       opencolorio
-      openimageio_2
+      openimageio
       opensubdiv
       ptex
       tbb

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -4098,9 +4098,9 @@ with pkgs;
   };
 
   osl = libsForQt5.callPackage ../development/compilers/osl {
-    libclang = llvmPackages_15.libclang;
-    clang = clang_15;
-    llvm = llvm_15;
+    libclang = llvmPackages_19.libclang;
+    clang = clang_19;
+    llvm = llvm_19;
   };
 
   ossec-agent = callPackage ../tools/security/ossec/agent.nix { };

